test/unit/math/mix/fun/eigen_comparisons_test.cpp

```diff
@@ -2,63 +2,83 @@
 #include <test/unit/util.hpp>
 #include <gtest/gtest.h>
 
-#define TEST_COMPARISON(SCALAR_TYPE, CONTAINER_TYPE1, CONTAINER_TYPE2, \
-                        CONTAINER1, CONTAINER2, OP)                    \
-  EXPECT_MATRIX_EQ(CONTAINER1 OP CONTAINER2,                           \
-                   CONTAINER_TYPE1(CONTAINER1)                         \
-                       OP CONTAINER_TYPE2(CONTAINER2));                \
-  EXPECT_MATRIX_EQ(1 OP CONTAINER2,                                    \
-                   SCALAR_TYPE(1) OP CONTAINER_TYPE2(CONTAINER2));     \
-  EXPECT_MATRIX_EQ(CONTAINER1 OP 1,                                    \
-                   CONTAINER_TYPE1(CONTAINER1) OP SCALAR_TYPE(1));
+template <typename Scalar, typename ContainerT1, typename ContainerT2,
+          typename Op, typename Container1Plain, typename Container2Plain>
+void test_comparison(Op operation, const Container1Plain& container1_plain,
+                     const Container2Plain& container2_plain) {
+  EXPECT_MATRIX_EQ(
+      operation(container1_plain, container2_plain),
+      operation(ContainerT1(container1_plain), ContainerT2(container2_plain)));
```

Comment:
Should mixes of containers be tested here since that's the new thing? Or are they being tested and I'm missing it?

```
operation(container1_plain, ContainerT2(container2_plain)));
```

Reply:
They are. See lines 25 and 27.

Reply:
Right, thanks!